rotkehlchen/db/upgrades/v31_v32.py

```diff
@@ -170,6 +170,12 @@ def _update_settings_name_for_selected_binance_markets(cursor: 'Cursor') -> None
     """, (BINANCE_MARKETS_KEY,))
 
 
+def _update_manual_balances_tags(cursor: 'Cursor') -> None:
+    manual_balances = cursor.execute('SELECT id, label FROM manually_tracked_balances').fetchall()
+    for balance_id, label in manual_balances:
```

Comment:
double iteration here again

```suggestion
    cursor.execute('SELECT id, label FROM manually_tracked_balances').fetchall()
    for balance_id, label in cursor:
```

Reply:
Maybe I misunderstand something but it doesn't work. This is what I was writing about in discord.

- If we leave `.fetchall()` (as your suggestion says), it will consume all data from the cursor and we won't iterate even once.
- If we remove `.fetchall()`, when we will make a new call to the cursor (inside the loop), cursor state will change and thus we will exit from the loop after the first iteration.

Reply:
ah sorry. Github rendered not what I meant. But don't you see the double iteration? Remove fetchall() and just use a second cursor (which you should also initialize outside of the loop).

Reply:
Done